src/replication.c

```diff
@@ -344,64 +349,80 @@ update_replica_list(int epfd, spec_t *spec, int replica_count) {
 	return replica_count;
 }
 
+void update_volstate(spec_t *spec) {
+	if(((spec->healthy_rcount + spec->degraded_rcount >= spec->consistency_factor) &&
+		(spec->healthy_rcount >= 1))||
+		(spec->healthy_rcount  + spec->degraded_rcount 
+			>= MAX(spec->replication_factor - spec->consistency_factor + 1, spec->consistency_factor))) {
+		spec->ready = true;
+		pthread_cond_broadcast(&spec->rq_cond);
+	} else {
+		spec->ready = false;
+	}
+}
+
+void clear_replica_cmd(spec_t *spec, replica_t *replica, rcmd_t *rep_cmd) {
+	int i;
+	rcommon_cmd_t *rcommq_ptr = rep_cmd->rcommq_ptr;
+	int luworker_id = rcommq_ptr->luworker_id;
+	rcommq_ptr->bitset &= ~(1 << replica->id);
```

Comment:
lock protection is required in using rcommon_cmd_t. created an issue #136 to make rcommon_cmd_t as thread safe.